File: articles/physical-chemistry-chemical-interpretation-matter/fortran/physical_chemistry_kernel.f90
program physical_chemistry_kernel
  implicit none

  real(8), parameter :: R = 8.314462618d0
  real(8), parameter :: F = 96485.33212d0
  real(8) :: K_demo, k_demo, E_demo

  K_demo = exp(-(-20.0d0 * 1000.0d0) / (R * 298.15d0))
  k_demo = 1.0d12 * exp(-(75.0d0 * 1000.0d0) / (R * 298.15d0))
  E_demo = 1.10d0 - (R * 298.15d0 / (2.0d0 * F)) * log(100.0d0)

  print *, "K_demo=", K_demo
  print *, "k_demo=", k_demo
  print *, "E_demo=", E_demo

end program physical_chemistry_kernel

File: articles/physical-chemistry-chemical-interpretation-matter/julia/physical_chemistry_kernel.jl
# Physical chemistry numerical kernel in Julia.

const R = 8.314462618
const F = 96485.33212
const kB = 1.380649e-23

function equilibrium_constant(delta_g_kj_mol, temperature_k)
    return exp(-(delta_g_kj_mol * 1000.0) / (R * temperature_k))
end

function arrhenius(A, Ea_kj_mol, temperature_k)
    return A * exp(-(Ea_kj_mol * 1000.0) / (R * temperature_k))
end

function nernst(E0, n, Q, T)
    return E0 - (R * T / (n * F)) * log(Q)
end

println("K_demo=", round(equilibrium_constant(-20.0, 298.15), digits=6))
println("k_demo=", arrhenius(1.0e12, 75.0, 298.15))
println("E_demo=", round(nernst(1.10, 2.0, 100.0, 298.15), digits=6))
